module/attribute/migrations/Version20201125121916.php

```diff
@@ -0,0 +1,22 @@
+<?php
+/*
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Migration;
+
+use Doctrine\DBAL\Schema\Schema;
+
+/**
+ * Auto-generated Ergonode Migration Class:
+ */
+final class Version20201125121916 extends AbstractErgonodeMigration
+{
+    public function up(Schema $schema): void
+    {
+        $this->addSql('ALTER TABLE attribute_option ALTER COLUMN "key" TYPE VARCHAR(128)');
```

Comment:
I think that this could be the potential location of the problem.
What if the current system has a key longer than 128 ? 
They will be trimmed but not in events and other projections. 

An additional two keys can be trimmed to the same one e.g:
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaazzzz (132 char)
and 
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaxxxx (132 char)
will be trimmed to 
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa (128 char)

If this be in the same attribute this migration will by bloked by database unique index restrictions:

`CREATE UNIQUE INDEX attribute_option_code_attribute_key ON attribute_option USING btree (attribute_id, key)`

Reply:
At the moment, the current length is checked at the entrance.

https://github.com/ergonode/backend/blob/bd834554e0838c0f082ff1b5fcedefdaeec44d51/module/attribute/src/Application/Model/Attribute/AttributeFormModel.php#L23